### PR TITLE
HELM-7: create an RPM package from the Helm app

### PR DESCRIPTION
**Pull request by RangerRick**
This PR adds a script to generate an RPM package in `dist/packages/`.  If you want to test, it works fine with Homebrew's RPM.

### CODE REVIEW

**Comment by RangerRick** [comment]
I've updated this to make the package not be 200MB of `node_modules`, and also a few other cleanup tweaks.  ;)

File: makerpm.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    console.log('Created ./%s', file);
+  });
+
+  var rpmbuilddir = path.join(os.tmpdir(), 'rpmbuild');
+  rimraf.sync(rpmbuilddir);
+  fs.mkdirSync(rpmbuilddir);
+  fs.mkdirSync(path.join(rpmbuilddir, 'SOURCES'));
+  fs.mkdirSync(path.join(rpmbuilddir, 'RPMS'));
+  fs.mkdirSync(path.join(rpmbuilddir, 'BUILD'));
+  fs.linkSync(path.join('SOURCES', pkginfo.name + '.tar.gz'), path.join(rpmbuilddir, 'SOURCES', pkginfo.name + '.tar.gz'));
```
Comment (RangerRick) [inline makerpm.js]: Ah, yeah, shouldn't be significantly different.  Fixed.

File: makerpm.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    'rpmbuild',
+    [
+      '--define', '_topdir ' + path.join(os.tmpdir(), 'rpmbuild'),
+      '-ba',
+      'SPECS/opennms-helm-app.spec'
+    ],
+    {
+      stdio: ['inherit', 'inherit', 'inherit']
+    }
+  );
```
Comment (RangerRick) [inline makerpm.js]: I added a `which` check, and then also exit(1) if the spawn fails.

**Comment by j-white** [comment]
My version of node (v6.11.1) doesn't seem to like that `catch` statement you added:

```
$ ./makerpm.js 
/home/jesse/git/grafana-opennms-helm-app/makerpm.js:30
catch (err) {
^^^^^
SyntaxError: Unexpected token catch
```

**Comment by RangerRick** [comment]
Haha yeah, turns out you're supposed to push *then* save in your editor... Should be fixed now.  Also fixed permissions to be owned by the `grafana` user.  (Should already be correct in the debian PR.)

**Comment by j-white** [comment]
Cool. I had just noticed that permission problem too. Let me give this another whirl.

**Comment by j-white** [comment]
I still get tons of these messages when installing:
```
warning: user opennms-helm-app does not exist - using root
warning: group opennms-helm-app does not exist - using root
```
Also, I don't think you meant to include the `opennms-helm-app` directory in this commit.

**Comment by RangerRick** [comment]
Yeah, it looks like somewhere deep inside, `specit` is always setting the username to the package name.  I just changed the template to hardcode it to `grafana`.  Also yeah, deleted the helm app directory.  Perils of switching back and forth between the RPM and debian branches.  :)

**Comment by j-white** [comment]
Permissions are good now, but I still have a problem with that catch statement:
```
$ ./makerpm.js 
/home/jesse/git/grafana-opennms-helm-app/makerpm.js:30
catch (err) {
^^^^^
SyntaxError: Unexpected token catch
```

**Comment by RangerRick** [comment]
Hm, I'm not sure how, since line 30 doesn't look like that anymore; it's got the `}` at the beginning.

**Comment by j-white** [comment]
Yep, my bad.

**Pull request merged**